python/ray/serve/_private/proxy.py

```diff
@@ -143,28 +149,23 @@ def __init__(
         node_ip_address: str,
         proxy_router_class: Type[ProxyRouter],
         request_timeout_s: Optional[float] = None,
-        controller_actor: Optional[ActorHandle] = None,
     ):
         self.request_timeout_s = request_timeout_s
         if self.request_timeout_s is not None and self.request_timeout_s < 0:
             self.request_timeout_s = None
 
         self._node_id = node_id
 
+        # Flipped to `True` once the route table has been updated at least once.
+        # Health checks will not pass until the route table is populated.
+        self._route_table_populated = False
+
         # Used only for displaying the route table.
-        self.route_info: Dict[str, EndpointTag] = dict()
+        self.route_info: Dict[str, DeploymentID] = dict()
```

Comment:
This type hint was just wrong :(

Reply:
It is a bit unnecessary to alias `DeploymentID` as `EndpointTag` but I don't think it's wrong tho. Might be useful if we ever change the type again 😅

Reply:
Oh geez -_- I didn't even see that. `EndpointTag` used to just be a str and I guess at some point it got swapped...